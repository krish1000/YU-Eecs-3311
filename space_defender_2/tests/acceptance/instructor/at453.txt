--Test enemy spawning does not occur due to being blocked.
toggle_debug_mode
play(5,30,1,1,40,40,40)
setup_select(2)
setup_next(5)

fire
fire
fire
pass
pass
