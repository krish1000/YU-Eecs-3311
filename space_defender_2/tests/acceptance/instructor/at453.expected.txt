  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,30,1,1,40,40,40)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(2)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Spread
->setup_next(5)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:120/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:110/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,30]
  Projectile:
    [-1,*]->damage:50, move:1, location:[B,2]
    [-2,*]->damage:50, move:1, location:[C,2]
    [-3,*]->damage:50, move:1, location:[D,2]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [C,1].
      A friendly projectile(id:-1) spawns at location [B,2].
      A friendly projectile(id:-2) spawns at location [C,2].
      A friendly projectile(id:-3) spawns at location [D,2].
  Enemy Action:
  Natural Enemy Spawn:
    A Carrier(id:1) spawns at location [A,30].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  C
    B _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:104/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,28]
    [2,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,30]
  Projectile:
    [-1,*]->damage:50, move:1, location:[A,3]
    [-2,*]->damage:50, move:1, location:[C,3]
    [-3,*]->damage:50, move:1, location:[E,3]
    [-4,*]->damage:50, move:1, location:[B,2]
    [-5,*]->damage:50, move:1, location:[C,2]
    [-6,*]->damage:50, move:1, location:[D,2]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [B,2] -> [A,3]
    A friendly projectile(id:-2) moves: [C,2] -> [C,3]
    A friendly projectile(id:-3) moves: [D,2] -> [E,3]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [C,1].
      A friendly projectile(id:-4) spawns at location [B,2].
      A friendly projectile(id:-5) spawns at location [C,2].
      A friendly projectile(id:-6) spawns at location [D,2].
  Enemy Action:
    A Carrier(id:1) moves: [A,30] -> [A,28]
  Natural Enemy Spawn:
    A Carrier(id:2) spawns at location [A,30].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  C  _  C
    B _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  *  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:98/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,26]
    [2,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,28]
  Projectile:
    [-2,*]->damage:50, move:1, location:[C,4]
    [-4,*]->damage:50, move:1, location:[A,3]
    [-5,*]->damage:50, move:1, location:[C,3]
    [-6,*]->damage:50, move:1, location:[E,3]
    [-7,*]->damage:50, move:1, location:[B,2]
    [-8,*]->damage:50, move:1, location:[C,2]
    [-9,*]->damage:50, move:1, location:[D,2]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,3] -> out of board
    A friendly projectile(id:-2) moves: [C,3] -> [C,4]
    A friendly projectile(id:-3) moves: [E,3] -> out of board
    A friendly projectile(id:-4) moves: [B,2] -> [A,3]
    A friendly projectile(id:-5) moves: [C,2] -> [C,3]
    A friendly projectile(id:-6) moves: [D,2] -> [E,3]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [C,1].
      A friendly projectile(id:-7) spawns at location [B,2].
      A friendly projectile(id:-8) spawns at location [C,2].
      A friendly projectile(id:-9) spawns at location [D,2].
  Enemy Action:
    A Carrier(id:1) moves: [A,28] -> [A,26]
    A Carrier(id:2) moves: [A,30] -> [A,28]
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  C  _  C  _  _
    B _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  *  *  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:106/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,24]
    [2,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,26]
    [4,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,24]
    [6,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,26]
    [7,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,30]
  Projectile:
    [-2,*]->damage:50, move:1, location:[C,5]
    [-5,*]->damage:50, move:1, location:[C,4]
    [-7,*]->damage:50, move:1, location:[A,3]
    [-8,*]->damage:50, move:1, location:[C,3]
    [-9,*]->damage:50, move:1, location:[E,3]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [C,4] -> [C,5]
    A friendly projectile(id:-4) moves: [A,3] -> out of board
    A friendly projectile(id:-5) moves: [C,3] -> [C,4]
    A friendly projectile(id:-6) moves: [E,3] -> out of board
    A friendly projectile(id:-7) moves: [B,2] -> [A,3]
    A friendly projectile(id:-8) moves: [C,2] -> [C,3]
    A friendly projectile(id:-9) moves: [D,2] -> [E,3]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [C,1], doubling regen rate.
  Enemy Action:
    A Carrier(id:1) moves: [A,26] -> [A,24]
      A Interceptor(id:3) spawns at location out of board.
      A Interceptor(id:4) spawns at location [B,24].
    A Carrier(id:2) moves: [A,28] -> [A,26]
      A Interceptor(id:5) spawns at location out of board.
      A Interceptor(id:6) spawns at location [B,26].
  Natural Enemy Spawn:
    A Carrier(id:7) spawns at location [B,30].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  C  _  C  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  I  _  I  _  _  _  C
    C S  _  *  *  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:114/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,22]
    [2,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,24]
    [4,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,23]
    [6,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,24]
    [7,C]->health:200/200, Regen:10, Armour:15, Vision:15, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,28]
    [9,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,22]
    [11,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,28]
    [12,I]->health:50/50, Regen:0, Armour:0, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[C,28]
  Projectile:
    [-2,*]->damage:50, move:1, location:[C,6]
    [-5,*]->damage:50, move:1, location:[C,5]
    [-8,*]->damage:50, move:1, location:[C,4]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [C,5] -> [C,6]
    A friendly projectile(id:-5) moves: [C,4] -> [C,5]
    A friendly projectile(id:-7) moves: [A,3] -> out of board
    A friendly projectile(id:-8) moves: [C,3] -> [C,4]
    A friendly projectile(id:-9) moves: [E,3] -> out of board
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [C,1], doubling regen rate.
  Enemy Action:
    A Carrier(id:1) moves: [A,24] -> [A,22]
      A Interceptor(id:8) spawns at location out of board.
      A Interceptor(id:9) spawns at location [B,22].
    A Carrier(id:2) moves: [A,26] -> [A,24]
      A Interceptor(id:10) spawns at location out of board.
    A Carrier(id:7) moves: [B,30] -> [B,28]
      A Interceptor(id:11) spawns at location [A,28].
      A Interceptor(id:12) spawns at location [C,28].
    A Interceptor(id:4) moves: [B,24] -> [B,23]
    A Interceptor(id:6) moves: [B,26] -> [B,24]
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  C  _  C  _  _  _  I  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  I  I  I  _  _  _  C  _  _
    C S  _  _  *  *  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  I  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
